tests/unit/plugins/test_rust_plugin.py

```diff
@@ -262,6 +262,78 @@ def callback_fail():
     ]
 
 
+@pytest.mark.parametrize("after", [["something-else"], []])
+def test_validate_environment_should_have_rustup(
+    after, fake_process: pytest_subprocess.FakeProcess, mocker
+):
+    def fake_execute(cmd: str):
+        return subprocess.check_output(  # noqa: S602
+            cmd,
+            shell=True,
+        )
+
+    def exec_fail(x):
+        raise subprocess.CalledProcessError(1, x)
+
+    fake_process.register(
+        ["cargo", "--version"], stdout="cargo 1.84.0-nightly (69e595908 2024-11-16)"
+    )
+    fake_process.register(
+        ["rustc", "--version"], stdout="rustc 1.84.0-nightly (5ec7d6eee 2024-11-17)"
+    )
+    fake_process.register(["rustup", "dump-testament"], callback=exec_fail)
+
+    mocker.patch(
+        "craft_parts.plugins.validator.PluginEnvironmentValidator._execute",
+        side_effect=fake_execute,
+    )
+    properties = RustPlugin.properties_class.unmarshal({"source": ".", "after": after})
+    validator = RustPlugin.validator_class(
+        part_name="my-part", properties=properties, env=""
+    )
+    # positive test
+    validator.validate_environment(part_dependencies=["rust-deps"])
+    # negative test
+    with pytest.raises(PluginEnvironmentValidationError):
+        validator.validate_environment(part_dependencies=after)
+
+
+def test_validate_environment_should_not_have_rustup(
+    fake_process: pytest_subprocess.FakeProcess, mocker
+):
+    def fake_execute(cmd: str):
+        return subprocess.check_output(  # noqa: S602
+            cmd,
+            shell=True,
+        )
+
+    def exec_fail(x):
+        raise subprocess.CalledProcessError(1, x)
+
+    fake_process.register(["cargo", "--version"], callback=exec_fail)
+    fake_process.register(["rustc", "--version"], callback=exec_fail)
+    fake_process.register(
+        ["rustup", "dump-testament"],
+        stdout="Rustup version renders as: 1.27.1 (2024-05-07)",
+    )
+
+    mocker.patch(
+        "craft_parts.plugins.validator.PluginEnvironmentValidator._execute",
+        side_effect=fake_execute,
+    )
+    properties = RustPlugin.properties_class.unmarshal(
+        {"source": ".", "after": ["rust-deps"]}
+    )
+    validator = RustPlugin.validator_class(
+        part_name="my-part", properties=properties, env=""
+    )
+    # positive test
+    validator.validate_environment(part_dependencies=[])
+    # negative test
+    with pytest.raises(PluginEnvironmentValidationError):
+        validator.validate_environment(part_dependencies=["rust-deps"])
```

Comment:
Please separate this into two test functions. The validator (and its dependencies) can go into a fixture, so the tests would only be a couple of lines each.